### PR TITLE
Extract NodeController from l7 lb

### PR DESCRIPTION
**Pull request by panslava**

### CODE REVIEW

File: cmd/glbc/main.go [hunk 1]
```diff
@@ -349,6 +351,10 @@ func runControllers(ctx *ingctx.ControllerContext) {
 
 	ctx.Start(stopCh)
 
+	instancePool := instances.NewNodePool(ctx.Cloud, ctx.ClusterNamer, ctx, utils.GetBasePath(ctx.Cloud))
```
Comment (kl52752) [inline cmd/glbc/main.go:354]: pass this instance Pool to L4NetLBcontroller and L7 (controller.go)

File: cmd/glbc/main.go [hunk 2]
```diff
@@ -349,6 +351,10 @@ func runControllers(ctx *ingctx.ControllerContext) {
...
+	nodeController := controller.NewNodeController(ctx, instancePool)
```
Comment (kl52752) [inline cmd/glbc/main.go:355]: I think that you should add health check function to context to check that this controller is working

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -319,16 +316,13 @@ func NewLoadBalancerController(
 
 // Init the controller
 func (lbc *LoadBalancerController) Init() {
-	// TODO(rramkumar): Try to get rid of this "Init".
```
Comment (panslava) [inline pkg/controller/controller.go:322]: I am not sure, was this comment referring about only first "Init" or not

File: cmd/glbc/main.go (hunk 2 above)
Comment (panslava) [reply cmd/glbc/main.go]: Looks like NodeController does not have any function show it healthieness to create healthCheck

I can create such a function in NodeController in the same way as it done in NegController, do you think it is needed?

**Pull request closed without merging**